parsl/probe/probe.py

```diff
@@ -0,0 +1,265 @@
+
+                logger.debug('\tThe provider (with launcher{wo}) returned with terminal status, and the env variables were consistent'.format(wo=' disabled' if disable_launcher else ''))
+                shutil.rmtree(os.path.expanduser('~/parsl_tmp'))
+                os.remove('./env.txt')
+
+            except Exception as e:
+                raise Exception(e)
+
+    def check_executor(self, func, arg, expected):
+
+        # This method can also check the executors. But the uncommented one
+        # is better, bacause it does not involve dfk.
+        #
+        # dfk = parsl.load(self.config)
+        # for label, executor in dfk.executors.items():
+        #     fut = executor.submit(square, None, 2)
+        #     print(fut.result())
+
+        logger.debug('\nBegin checking the executor(s) in the config')
+        executors = self.config.executors
+        logger.debug(f'There are {len(executors)} executors in the config')
+
+        avail_spec = {'cores': 2, 'memory': 1000, 'disk': 1000}
+        for i, executor in enumerate(executors):
+            logger.debug(f'\tBegin checking #{i+1} executor in the config')
+            executor.provider.script_dir = '.'
+            executor.provider.channel.script_dir = '.'
+            block_id = executor.start()
+            spec = None
+            if type(executor) is WorkQueueExecutor:
+                spec = avail_spec
+            fut = executor.submit(func, spec, arg)
+            try:
+                if fut.result() == expected:
+                    logger.debug(f'\t#{i+1} executor works properly')
+                else:
+                    logger.debug(f'\t#{i+1} executor does NOT work properly')
+            except Exception as e:
+                raise Exception(f'\t#{i+1} executor met exception {e}')
+
+
+def start_logger(filename, rank, name='parsl', level=logging.DEBUG, format_string=None):
+    """Add a file handler and a stream handler.
+
+    Args:
+        - filename (string): Name of the file to write logs to
+        - name (string): Logger name
+        - level (logging.LEVEL): Set the logging level.
+        - format_string (string): Set the format string
+
+    Returns:
+       -  None
+    """
+    if format_string is None:
+        format_string = "%(asctime)s.%(msecs)03d %(name)s:%(lineno)d Rank:{0} [%(levelname)s]  %(message)s".format(rank)
+    formatter = logging.Formatter(format_string, datefmt='%Y-%m-%d %H:%M:%S')
+
+    global logger
+    logger = logging.getLogger(name)
+    logger.setLevel(logging.DEBUG)
+
+    file_handler = logging.FileHandler(filename)
+    file_handler.setLevel(level)
+    file_handler.setFormatter(formatter)
+    logger.addHandle(file_handler)
+
+    stream_handler = logging.StreamHandler()
+    stream_handler.setLevel(level)
+    stream_handler.setFormatter(formatter)
+    logger.addHandle(stream_handler)
+
+
+def cli_run():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("config_file")
```

Comment:
Make sure to give help lines. Might be useful to look at the style here-> https://github.com/Parsl/parsl/blob/master/parsl/executors/high_throughput/process_worker_pool.py#L594